nsd_android/android/src/main/kotlin/com/haberey/flutter/nsd_android/NsdAndroidPlugin.kt

```diff
@@ -85,14 +86,10 @@ class NsdAndroidPlugin : FlutterPlugin, MethodCallHandler {
             "Cannot start discovery: expected handle"
         )
 
-        if (multicastLock == null) {
-            throw NsdError(
-                ErrorCause.SECURITY_ISSUE,
-                "Missing required permission CHANGE_WIFI_MULTICAST_STATE"
-            );
-        }
-
-        multicastLock?.acquire()
+        multicastLock?.acquire() ?: throw NsdError(
```

Comment:
Not sure about this one. These are two distinct error scenarios: 

1. failure to create multicast lock - points to missing permission CHANGE_WIFI_MULTICAST_STATE 
1. failure to acquire multicast lock - I don't exactly know when this condition occurs, but most likely does NOT mean missing permission

IMO the error message would be misleading in case 2.

Reply:
Oh. I assumed acquire() had a return value and that was why you placed the throw clause behind it.

Since it doesn't, all the change does is make a casual reader assume the error is related to the acquire() statement.

Reply:
Sorry, this PR was closed due to my mistake. I opened a new PR. Please check！